frontend/src/components/LoginForm.tsx

```diff
@@ -66,7 +67,7 @@ const LoginForm = (props: any) => {
       clearErrors();
       clearModals();
       // redirect to profile page
-      history.push('/ProfilePage');
+      if (location.pathname === '/Login') history.push('/ProfilePage');
```

Comment:
Probably a good idea to include " || location.pathname ===  '/signup' " in there, or just (location.pathname !== '/VisualCourse') . Otherwise LGTM

Reply:
Agree with adding a check, but in SignupForm, for the isAuthed redirect

Reply:
i was thinking about that too but i figured that when you first signup maybe you would always get redirected to the profile page? idk

Reply:
Sounds reasonable. Then should be documented in code as well